### DISCUSSION TITLE
Auto-complete support for Jedi/Pylance for writing code faster.

### DISCUSSION THREAD

**Discussion by astrabound** [pallets-eco/flask-sqlalchemy#1222] [Q&A] [answered]
I have been using flask for a while, and default to this extension for database integration.

One of the issues I've been facing is:
![image](https://github.com/pallets-eco/flask-sqlalchemy/assets/117508601/4041bcf6-6266-4737-a955-f580ac427b8a)

Auto complete support being missing for the package really hurts when there are already a lot of things to remember for the complex apps that are using it.

I feel this can be an even powerful tool with the same.

I went through Jedi's issue - https://github.com/davidhalter/jedi/issues/1372

Is this something that can be implemented? I am willing to help with the same.

**Comment by pamelafox** [comment, marked as answer]
There's a proposed fix here:
https://github.com/pallets-eco/flask-sqlalchemy/pull/1207

I just tested it out myself by replacing Flask-SQLAlchemy in my requirements.txt with this line:
git+https://github.com/Zoupers/flask-sqlalchemy.git@0a2af2fd9bee6dd7ccdb628caa065d30ddb14613

### Before: (Current Flask-SQLAlchemy 3.0.3)
<img width="525" alt="Screenshot 2023-06-18 at 5 53 18 PM" src="https://github.com/pallets-eco/flask-sqlalchemy/assets/297042/20dfd3ac-eb76-45ee-acf9-4a18c16e03e9">


### After: (With the PR change)

<img width="627" alt="Screenshot 2023-06-18 at 5 55 07 PM" src="https://github.com/pallets-eco/flask-sqlalchemy/assets/297042/30b2df3b-03f8-4560-b769-038685fc50b0">

<img width="677" alt="Screenshot 2023-06-18 at 5 55 16 PM" src="https://github.com/pallets-eco/flask-sqlalchemy/assets/297042/957b7862-0881-4902-b41c-231d5a104a7a">

Feel free to try it and see if it works for you as well. We hope to release the fix soon.

**Comment by yomajo** [comment]
> There's a proposed fix here: #1207
> [...]

Does this add suggestions to other things?
![image](https://github.com/pallets-eco/flask-sqlalchemy/assets/45366313/95b17be2-8ad3-4381-bcaa-01d18c59668e)

- `db.session.` > scalars/scalar
- `db.` > select
- `db.select.` > where/order_by/join/filter..
- `db.session.scalars.` > first/one/all/..

**Reply by astrabound** [reply]
@yomajo @pamelafox this is why I love open-source :') Thanks for the quick resolution.
